### PR TITLE
Improve the default client loading code.

### PR DESCRIPTION
**Pull request by brendandburns**

### CODE REVIEW

File: src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs [hunk 1]
```diff
@@ -24,6 +24,27 @@ public partial class KubernetesClientConfiguration
...
+        /// </summary>
+        public static KubernetesClientConfiguration BuildDefaultConfig() {
+            var kubeconfig = Environment.GetEnvironmentVariable("KUBECONFIG");
+            if (kubeconfig != null) {
+                return BuildConfigFromConfigFile(kubeconfigPath: kubeconfig);
+            }
+            if (File.Exists(KubeConfigDefaultLocation)) {
+                return BuildConfigFromConfigFile(kubeconfigPath: KubeConfigDefaultLocation);
+            }
+            var tokenPath = KubernetesClientConfiguration.ServiceaccountPath + KubernetesClientConfiguration.ServiceAccountTokenKeyFileName;
```
Comment (itowlson) [inline src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs]: Prefer using `Path.Combine` to build directory paths.

File: src/KubernetesClient/KubernetesClientConfiguration.InCluster.cs [hunk 2]
```diff
@@ -6,9 +6,9 @@ namespace k8s
 {
     public partial class KubernetesClientConfiguration
     {
-        private const string ServiceaccountPath = "/var/run/secrets/kubernetes.io/serviceaccount/";
-        private const string ServiceAccountTokenKeyFileName = "token";
-        private const string ServiceAccountRootCAKeyFileName = "ca.crt";
+        public const string ServiceaccountPath = "/var/run/secrets/kubernetes.io/serviceaccount/";
```
Comment (itowlson) [inline src/KubernetesClient/KubernetesClientConfiguration.InCluster.cs]: `Service*A*ccountPath` (capital A)

File: src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs [hunk 3]
```diff
@@ -24,6 +24,27 @@ public partial class KubernetesClientConfiguration
...
+            if (File.Exists(tokenPath)) {
```
Comment (tg123) [inline src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs]: Here should be conform with the logic within `InClusterConfig`

Maybe it is better to extract `IsInCluster ` from  `InClusterConfig` and change `InClusterConfig` to `assert IsInCluster`

```
IsInCluster => {
            var host = Environment.GetEnvironmentVariable("KUBERNETES_SERVICE_HOST");
            var port = Environment.GetEnvironmentVariable("KUBERNETES_SERVICE_PORT");

            return (string.IsNullOrWhiteSpace(host) || string.IsNullOrWhiteSpace(port)) ;
}
```

File: src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs (hunk 1 above)
Comment (brendandburns) [reply src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs]: done.

File: src/KubernetesClient/KubernetesClientConfiguration.InCluster.cs (hunk 2 above)
Comment (brendandburns) [reply src/KubernetesClient/KubernetesClientConfiguration.InCluster.cs]: done.

File: src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs (hunk 3 above)
Comment (brendandburns) [reply src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs]: done.

**Comment by brendandburns** [comment]
Comments addressed, please take another look.

**Pull request merged**